### PR TITLE
chore(cketh/ckerc20): proposal to upgrade the ckETH minter to support subaccounts

### PR DESCRIPTION
**Pull request by gregorydemay**

### CODE REVIEW

File: rs/ethereum/cketh/mainnet/minter_upgrade_2024_11_22.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    * The previous helper smart contracts are marked as deprecated since the new contract can handle all types of deposits supported by the minter. Future proposals may decide to reduce the frequency at which the logs of the deprecated helper smart contracts are being scraped.
+* for withdrawals: subaccounts can be specified when withdrawing ckETH or ckERC20.
+
+## Upgrade args
+
+```
+git fetch
+git checkout 2181ddf2a690ca0262d2d9d0511b093bfa350ece
+cd rs/ethereum/cketh/minter
+didc encode -d cketh_minter.did -t '(MinterArg)' '(variant {UpgradeArg = record {deposit_with_subaccount_helper_contract_address = opt "0x18901044688D3756C35Ed2b36D93e6a5B8e00E68"; last_deposit_with_subaccount_scraped_block_number = opt 21241175}})' | xxd -r -p | sha256sum
```
Comment (gregorydemay) [inline rs/ethereum/cketh/mainnet/minter_upgrade_2024_11_22.md:32]: `242a21d3e36313843547a0620cba9a45b194ad46c7ceb61fb16f92231b2a8652`

Comment (THLO) [reply rs/ethereum/cketh/mainnet/minter_upgrade_2024_11_22.md:32]: Confirmed!

File: rs/ethereum/cketh/mainnet/minter_upgrade_2024_11_22.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+## Wasm Verification
+
+Verify that the hash of the gzipped WASM matches the proposed hash.
+
+```
+git fetch
+git checkout 2181ddf2a690ca0262d2d9d0511b093bfa350ece
+"./ci/container/build-ic.sh" "--canisters"
+sha256sum ./artifacts/canisters/ic-cketh-minter.wasm.gz
```
Comment (THLO) [inline rs/ethereum/cketh/mainnet/minter_upgrade_2024_11_22.md:64]:
```
$ sha256sum ./artifacts/canisters/ic-cketh-minter.wasm.gz
47794d7e689c21377b4589f7b304f0d9b8113896b85347aa4e116629cd0941e5  ./artifacts/canisters/ic-cketh-minter.wasm.gz
```
:+1:

**Review by THLO — APPROVED** [review]

File: rs/ethereum/cketh/mainnet/minter_upgrade_2024_11_22.md (hunk 2 above)
Comment (ninegua) [reply rs/ethereum/cketh/mainnet/minter_upgrade_2024_11_22.md:64]: Hash confirmed!

File: rs/ethereum/cketh/mainnet/minter_upgrade_2024_11_22.md (hunk 1 above)
Comment (ninegua) [reply rs/ethereum/cketh/mainnet/minter_upgrade_2024_11_22.md:32]: Confirmed!

**Review by ninegua — APPROVED** [review]

**Pull request merged**